libcloud/common/base.py

```diff
@@ -46,11 +48,48 @@
 
 from libcloud.utils.misc import lowercase_keys
 from libcloud.utils.compression import decompress_data
+from libcloud.common.exception import RateLimit, exception_from_message
 from libcloud.common.types import LibcloudError, MalformedResponseError
-
 from libcloud.httplib_ssl import LibcloudHTTPConnection
 from libcloud.httplib_ssl import LibcloudHTTPSConnection
 
+DEFAULT_TIMEOUT = 3
+DEFAULT_SLEEP = 3
+DEFAULT_BACKCOFF = 1
+
+
+def retry(*dargs, **dkwargs):
```

Comment:
As we discussed IRL, this would be better as a function that you could configure those values at runtime.  I opted to call ```sleep``` ```retry_delay``` because that is more obvious how it's used.  The usage would be something like this in the request method:

```
retry(self.connection.timeout, self.connection.retry_delay, self.connection.request, method=method, url=url, body=data, headers=headers)
```

And you'd need to modify ```libcloud.common.base.BaseDriver.__init__``` to take those parameters and pass them along to the connection class:

```
conn_kwargs = self._ex_connection_class_kwargs()
conn_kwargs.update({'timeout': kwargs.pop('timeout'), 'retry_delay': kwargs.pop('retry_delay')})
self.connection = self.connectionCls(*args, **conn_kwargs)
```

As well as modify the ```Connection``` class to take the ```retry_delay``` parameter and store it.  Interestingly it already has a ```timeout``` property, but it just isn't passed along from the driver for some reason, as far as I can tell, so we'll just have to do that part as described above.